src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -387,65 +366,42 @@ public Color mixWithDyes(DyeColor... dyeColors) {
         return mixColors(newColorArray);
     }
 
-    @Override
-    public int getContentVersion() {
-        return 1;
-    }
-
-    @Override
-    public DataContainer toContainer() {
-        return DataContainer.createNew()
-                .set(Queries.CONTENT_VERSION, this.getContentVersion())
-                .set(Queries.COLOR_RED, this.getRed())
-                .set(Queries.COLOR_GREEN, this.getGreen())
-                .set(Queries.COLOR_BLUE, this.getBlue());
-    }
-
-    @Override
-    public int hashCode() {
-        return this.rgb;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        return this == obj || obj != null && getClass() == obj.getClass() && this.rgb == ((Color) obj).rgb;
-    }
+    interface Builder extends ResettableBuilder<Color, Builder> {
 
-    @Override
-    public String toString() {
-        return com.google.common.base.MoreObjects.toStringHelper(this)
-                .add("rgb", this.rgb)
-                .add("red", this.red)
-                .add("green", this.green)
-                .add("blue", this.blue)
-                .toString();
-    }
+        /**
+         * Sets the rgb value for this color. An alpha value will be stripped.
+         *
+         * @param rgb The red-green-blue value
+         * @return This builder
+         */
+        Builder rgb(int rgb);
 
-    public static final class Builder extends AbstractDataBuilder<Color> {
+        /**
+         * Sets the rgb value for this color using red, green, and blue values.
+         * Values outside the range of 0 through 255 (0xFF) inclusive will be
+         * brought in range using {@code value & 0xFF}.
+         *
+         * @param red The red value
+         * @return This builder
+         */
+        Builder rgb(int red, int green, int blue);
 
         /**
-         * Creates a new {@link Builder} for building {@link Color} objects,
-         * either from {@link DataView}s, or otherwise.
+         * Sets the red, green, and blue values for this color by taking the
+         * average of each for every color in the array.
+         *
+         * @param colors The colors to mix
+         * @return This builder
          */
-        public Builder() {
-            super(Color.class, 1);
-        }
+        Builder mix(Color... colors);
```

Comment:
A better pattern here is that mix takes one color and mixes with the current color of the builder

Reply:
That would have default implementations mix with the color black. Any static method for mixing would need to pull out the first element of the array, build thay, then copy the remaining elements to a new array to be used with this method. The mix methods aren't used in Sponge (bar tests) and are probably used in few to no plugins. I'd be more in favor of removing them entirely.